tests/extension/khr_free_function_commands/free_function_commands.cpp

```diff
@@ -0,0 +1,657 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2025 The Khronos Group Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+
+#include "../../common/common.h"
+
+namespace free_function_commands::tests {
+
+#ifdef SYCL_KHR_FREE_FUNCTION_COMMANDS
+static void test_submit() {
+  sycl::queue q;
```

Comment:
Will it respect CTS device selector (i.e. the ability to specify device though command line option)? I think that we should use utilities from `get_cts_object` instead:

https://github.com/KhronosGroup/SYCL-CTS/blob/ef8b127dddcfa59a50781c5a441ad56724aafa32/tests/common/get_cts_object.h#L77-L86